### ISSUE TITLE
TypeError: r.getAllMarks

### ISSUE THREAD

**Issue by channingwalton** [marcusolsson/obsidian-vale#7]
Obisidian version v0.13.0

```
Uncaught TypeError: r.getAllMarks is not a function
    at eval (eval at <anonymous> (app.js:1), <anonymous>:367:59524)
    at Y8.withCodeMirrorEditor (eval at <anonymous> (app.js:1), <anonymous>:367:60241)
```

**Comment by marcusolsson** [comment]
Thanks for reporting! I'll try to look into this later this week.

**Comment by kmaustral** [comment]
Same here on v0.13.6
![image](https://user-images.githubusercontent.com/73867063/144320330-5ff8aef7-1590-448c-bf79-fabfd3f129c2.png)

**Comment by vincentteoh** [comment]
Same error message here. 
![image](https://user-images.githubusercontent.com/10986305/144356127-f02d6667-126b-4a6b-a25b-15bd2fe2bf42.png)

**Comment by marcusolsson** [comment]
Seems like the API has changed in v13.x. I'll have time to look at this next week. Thanks for reporting this!

**Comment by marcusolsson** [comment]
I believe that once Obsidian 13 is officially released, it'll have a Legacy Editor option to support plugins that depend on features that haven't yet been made available to plugins in 13.x. I'll look for a proper fix, but at least the plugin will continue to work with the new option.

**Comment by marcusolsson** [comment]
I've added a Troubleshooting section in the README on how to enable the Legacy Editor for now.

**Comment by rushikb** [comment]
Hi Marcus! I'm a big fan of Vale— just wanted to check whether you still plan on supporting CM6 now that Obsidian 13 is officially released. This is such a phenomenal tool

**Comment by marcusolsson** [comment]
Hey! I'm definitely planning to upgrade the plugin. I haven't had the chance to look into it yet, but AFAIK the new Editor API has yet to achieve feature parity with CM5, and doesn't support markers yet. If there's a way to grab the CM6 instance somewhere, that might also work. If anyone knows how, I'd be happy to prioritize this. Otherwise, I'll try and find some time for this in the coming weeks.

**Comment by chrisgrieser** [comment]
yeah I came here to report the same thing – it seems the new editor is not supported :(

<img width="340" alt="Screenshot 2021-12-22 21 23 53" src="https://user-images.githubusercontent.com/73286100/147151274-1acd5d52-edee-4c4d-ac54-328e6359314b.png">

**Comment by marcusolsson** [comment]
@chrisgrieser Sorry 'bout that 🙏 I'll try and look into this next week.

**Comment by marcusolsson** [comment]
Just to give an update. I've been reading up on how to migrate from CM5 to CM6, and unfortunately there doesn't seem to be an easy fix to this. 

If anyone want to experiment, here's the best resources I've found so far:

- [CM6 example](https://github.com/nothingislost/obsidian-cm6-attributes)
- [Extending CodeMirror](https://codemirror.net/6/docs/guide/#extending-codemirror)
- [Marked text](https://codemirror.net/6/docs/migration/#marked-text)

Seems like we'd need to build a CodeMirror extension rather than calling methods for adding marks. 

I haven't given up on this, but right now this seems to require more time than I have at the moment. Seems like I need to read the CodeMirror docs more closely before I give this another try.

Thanks for the patience! 🙏

**Comment by eleanorkonik** [comment]
Thanks for including the experimentation tips! I'll include this in the Roundup :)

**Comment by mohitmujawdiya** [comment]
Hi! Thank you for creating such an awesome extension. While it works like a breeze in the legacy editor, I simply can't leave the features of the new editor like a live preview. Any update on this?

**Comment by marcusolsson** [comment]
Thank you @mohitmujawdiya! No updates unfortunately. 

I definitely want to fix this, though I haven't yet found the time to work on it. I don't have any estimate on when I'll be able to.

**Comment by mohitmujawdiya** [comment]
hi Marcus! no problem, i'll wait. (will keep using linter and language tool plugins for now)
